src/tensors/cpu/intgemm_interface.h

```diff
@@ -0,0 +1,340 @@
+#pragma once
+
+#include "graph/node.h"
+#include "graph/node_operators_unary.h"
+#include "integer_common.h"
+
+namespace marian {
+
+namespace cpu {
+namespace integer {
+
+template<Type vtype>
+struct PrepareANodeOp : public NaryNodeOp {
+float clipValue_;
+float quantMult_;
+  PrepareANodeOp(Expr input, Expr quant_mult, float clipValue)
+      : NaryNodeOp({input, quant_mult}, input->shape(), vtype), clipValue_{clipValue} {
+
+    set_name(input->name());
+    setMemoize(false);
+    // Check if arguments are not null
+    ABORT_IF(child(0) == nullptr, "A cannot be null");
+    ABORT_IF(child(1) == nullptr, "Quant mult of A cannot be null");
+  }
+
+  NodeOps forwardOps() override {
+    return {NodeOp(
+      quantMult_ = *child(1)->val()->data();
+      typedef typename intgemm_<vtype>::type Integer;
+      intgemm_<vtype>::width::PrepareA(child(0)->val()->data(), /*input*/
+                                    val_->data<Integer>(), /*output*/
+                                    *child(1)->val()->data(), /*Quant Mult*/
+                                    rows(child(0)->val()),
+                                    cols(child(0)->val()));
+    )};
+  }
+
+  NodeOps backwardOps() override {
+    ABORT("Only used for inference");
+    return {NodeOp(0)};
+  }
+
+  const std::string type() override { return "intgemmPrepareA"; }
+};
+
+template<Type vtype>
+struct PrepareBNodeOp : public NaryNodeOp {
+float clipValue_;
+float quantMult_;
+
+  PrepareBNodeOp(Expr input, Expr quant_mult, float clipValue)
+      : NaryNodeOp({input, quant_mult}, input->shape(), intgemm_<vtype>::intgemmType), clipValue_{clipValue} {
+
+    set_name(input->name());
+    // Check if arguments are not null
+    ABORT_IF(child(0) == nullptr, "A cannot be null");
+    ABORT_IF(child(1) == nullptr, "Quant mult of B cannot be null");
+    ABORT_IF(input->shape()[-1] %8 != 0, "Columns of matrix: " + input->type() + " must be multiple of 8.");
```

Comment:
@XapaJIaMnu It would be nice to have a support for non 8 multiple matrices. We have many models with non 8 multiple vocab sizes.

Reply:
I know.  It's coming.  https://github.com/kpu/intgemm/blob/static/tile/dot.inl#L147